usersync/ejector.go

```diff
@@ -0,0 +1,146 @@
+package usersync
+
+import (
+	"errors"
+	"math"
+	"math/rand"
+	"time"
+)
+
+type Ejector interface {
+	Choose(uids map[string]UIDEntry) (string, error)
```

Comment:
I'd love to bring this up during team time to see what people think about a name change. The idea is for this function to "Choose" which uid to eject, but I'm always up for a name change!

Reply:
`Chooser` would better follow Go conventions, but we need to provide context as to what we're choosing because there are other similar things in the usersync package for choosing syncs. I like that Ejector provides that context. Maybe something like EjectionChooser or DeletionChooser?

Reply:
This may be the last issue remaining, getting the name right. I suppose the other solution would be to go:
```
type Ejector interface {
        Eject(uids map[string]UIDEntry) (string,error)
}
```

Reply:
Right, but this function doesn't Eject. It selects/chooses a uid to be ejected.

Reply:
I think expanding the name to be `EjectionChoose` is a good option to provide more context.